conformance_pack/iam.sp

```diff
@@ -451,32 +451,875 @@ control "iam_user_with_administrator_access_mfa_enabled" {
+      -- Additional Dimensions
+      ${local.tag_dimensions_sql}
+      ${replace(local.common_dimensions_qualifier_global_sql, "__QUALIFIER__", "p.")}
+    from
+      aws_iam_policy as p
+      left join bad_policies as bad on p.arn = bad.arn
+    where
+      not p.is_aws_managed;
+  EOQ
+}
 
-  tags = merge(local.conformance_pack_iam_common_tags, {
-    other_checks = "true"
-  })
+query "iam_root_user_no_access_keys" {
+  sql = <<-EOQ
+    select
+      -- Required Columns
+      'arn:' || partition || ':::' || account_id as resource,
+      case
+        when account_access_keys_present > 0 then 'alarm'
+        else 'ok'
+      end status,
+      case
+        when account_access_keys_present > 0 then 'Root user access keys exist.'
+        else 'No root user access keys exist.'
+      end reason
+      -- Additional Dimensions
+      ${local.tag_dimensions_sql}
+      ${replace(local.common_dimensions_qualifier_global_sql, "__QUALIFIER__", "i.")}
+    from
+      aws_iam_account_summary as i;
+  EOQ
 }
 
-control "iam_user_with_administrator_access_mfa_enabled" {
-  title       = "IAM administrator users should have MFA enabled"
-  description = "Manage access to resources in the AWS Cloud by ensuring MFA is enabled for users with administrative privileges."
-  query       = query.iam_user_with_administrator_access_mfa_enabled
+query "iam_root_user_hardware_mfa_enabled" {
+  sql = <<-EOQ
+    select
+      -- Required Columns
+      'arn:' || s.partition || ':::' || s.account_id as resource,
+      case
+        when account_mfa_enabled and serial_number is null then 'ok'
+        else 'alarm'
+      end status,
+      case
+        when account_mfa_enabled = false then  'MFA not enabled for root account.'
+        when serial_number is not null then 'MFA enabled for root account, but the MFA associated is a virtual device.'
+        else 'Hardware MFA device enabled for root account.'
+      end reason
+      -- Additional Dimensions
+      ${local.tag_dimensions_sql}
+      ${replace(local.common_dimensions_qualifier_global_sql, "__QUALIFIER__", "s.")}
+    from
+      aws_iam_account_summary as s
+      left join aws_iam_virtual_mfa_device on serial_number = 'arn:' || s.partition || ':iam::' || s.account_id || ':mfa/root-account-mfa-device';
+  EOQ
+}
 
-  tags = merge(local.conformance_pack_iam_common_tags, {
-    other_checks = "true"
-  })
+query "iam_root_user_mfa_enabled" {
+  sql = <<-EOQ
+    select
+      -- Required Columns
+      'arn:' || partition || ':::' || account_id as resource,
+      case
+        when account_mfa_enabled then 'ok'
+        else 'alarm'
+      end status,
+      case
+        when account_mfa_enabled then 'MFA enabled for root account.'
+        else 'MFA not enabled for root account.'
+      end reason
+      -- Additional Dimensions
+      ${local.tag_dimensions_sql}
+      ${replace(local.common_dimensions_qualifier_global_sql, "__QUALIFIER__", "i.")}
```

Comment:
Qualifier is not required here, can you please remove and validate for other queries as well .